core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -689,6 +689,18 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     new ParallelCollectionRDD[T](this, seq, numSlices, Map[Int, Seq[String]]())
   }
 
+  /** Generate Long data from 0 until `limit` to form an RDD.
+    *
+    * @note even we can set `limit` larger than Int.MaxValue, we need to make sure after partition
+    * each partition would have data size acceptable by setting a proper `numSlice`.
+    */
+  def range(
+      limit: Long,
+      numSlices: Int = defaultParallelism): RDD[Long] = withScope {
```

Comment:
I think we can implement a better `range` method, such as `def range(start: Long, end: Long, step: Long = 1, numSlices: Int = defaultParallelism)`. 

> we need to make sure after partition each partition would have data size acceptable by setting a proper `numSlice`.

Could you elaborate `data size acceptable`?